Update the event stream section on the RFC30. (#2243)
awslabs/smithy-rs: design/src/rfcs/rfc0030_serialization_and_deserialization.md

For the reasons mentioned above, we implement an enum that is externally tagged.

## Data Types to Skip Serialization/Deserialization
We are going to skip serialization and deserialization of fields that have the datatype that corresponds to `@streaming blob` from smithy.
Any fields with these data types are tagged with `#[serde(skip)]`.

> By skipping, corresponding field's value will be assined the value generated by `Default` trait.

Reviewer: ```suggestion
By skipping, corresponding field's value will be assigned the value generated by `Default` trait.
```

As of writing, `aws_smithy_http::byte_stream::ByteStream` is the only data type that is affected by this decision.

Here is an example of data types affected by this decision:
- `aws_sdk_s3::input::put_object_input::PutObjectInput`

We considered serializing them as bytes, however, it could take some time for a stream to reach the end, and the resulting serialized data may be too big for itself to fit into the ram.

Here is an example snippet.
```rust
#[allow(missing_docs)]
#[cfg_attr(
    all(aws-sdk-unstable, feature = "serde-serialize"),
    derive(serde::Serialize)
)]
#[cfg_attr(
    all(aws-sdk-unstable, feature = "serde-deserialize"),
    derive(serde::Deserialize)
)]
#[non_exhaustive]
#[derive(std::fmt::Debug)]
pub struct PutObjectInput {
    pub acl: std::option::Option<crate::model::ObjectCannedAcl>,
    pub body: aws_smithy_http::byte_stream::ByteStream,
    // ... other fields
}
```

> ## SDK Data types to skip

Reviewer: ```suggestion
## Data types to exclude from ser/de code generation
```

> For SDK data types that includes `@streaming union` in one of their field, we do NOT implement `serde` traits.

Reviewer: ```suggestion
For data types that include `@streaming union` in any of their fields, we do NOT implement `serde` traits.
```

> As of writing, following Rust data types corresponds to `@stremaing union`.

Reviewer: ```suggestion
As of writing, following Rust data types corresponds to `@streaming union`.
```
